src/Microsoft.ML.OnnxTransform/OnnxUtils.cs

```diff
@@ -0,0 +1,302 @@
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Scoring;
+using OnnxShape = System.Collections.Generic.List<long>;
+
+namespace Microsoft.ML.OnnxScoring
+{
+
+    /// <summary>
+    /// OnnxModel is a facad for ModelManager. ModelManager is provided by Sonoma API,
+    /// and it has a lot of functionality (multiple models, multiple versions) that are not
+    /// needed by Onnx transform, which only needs a single model. This facad simplifies the
+    /// usage of onnx model.
+    /// </summary>
+    internal sealed class OnnxModel
+    {
+        /// <summary>
+        /// OnnxModelInfo contains the data that we should get from
+        /// Sonoma API once that functionality is added.
+        /// </summary>
+        public sealed class OnnxModelInfo
+        {
+            public readonly OnnxNodeInfo[] InputsInfo;
+            public readonly OnnxNodeInfo[] OutputsInfo;
+
+            public OnnxModelInfo(OnnxNodeInfo[] inputsInfo, OnnxNodeInfo[] outputsInfo)
+            {
+                InputsInfo = inputsInfo;
+                OutputsInfo = outputsInfo;
+            }
+        }
+
+        /// <summary>
+        /// OnnxNodeInfo contains all the information for a given node (e.g. inputs/outputs)
+        /// of an Onnx model.
+        /// </summary>
+        public class OnnxNodeInfo
+        {
+            public readonly string Name;
+            public readonly OnnxShape Shape;
+            public readonly DataType Type;
+
+            public OnnxNodeInfo(string name, OnnxShape shape, DataType type)
+            {
+                Name = name;
+                Shape = shape;
+                Type = type;
+            }
+        }
+
+        public readonly OnnxModelInfo ModelInfo;
+
+        private static readonly int _ignoredVersion = int.MaxValue;
+        private readonly ModelManager _modelManager;
+        private readonly string _modelFile;
+        private readonly string _modelName;
+        private readonly List<string> _inputNames;
+        private readonly List<string> _outputNames;
+
+        public OnnxModel(string modelFile)
+        {
+            _modelFile = modelFile;
+
+            // Load the onnx model
+            var modelFileInfo = new FileInfo(modelFile);
+            _modelName = Path.GetFileNameWithoutExtension(modelFileInfo.Name);
+            _modelManager = new ModelManager(modelFileInfo.Directory.FullName, true);
+            _modelManager.InitOnnxModel(_modelName, _ignoredVersion);
+
+            ModelInfo = new OnnxModelInfo(GetInputsInfo(), GetOutputsInfo());
+            _inputNames = ModelInfo.InputsInfo.Select(i => i.Name).ToList();
+            _outputNames = ModelInfo.OutputsInfo.Select(i => i.Name).ToList();
+        }
+
+        public static OnnxModel CreateFromBytes(byte[] modelBytes)
+        {
+            var tempModelDir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            Directory.CreateDirectory(tempModelDir);
```

Comment:
>tempModelDir [](start = 38, length = 12)

might consider ACL this temp dir . 

Temp directories with models are considered executable code and considered security threat by .NET security team